doc/coq.rst

```diff
@@ -142,19 +142,10 @@ The semantics of the fields are:
   You may still use installed libraries in your Coq project, but there is
   currently no way for Dune to know about it.
 
-- You can enable the production of Coq's native compiler object files by setting
-  ``<coq_native_mode>`` to ``native``. This passes ``-native-compiler on`` to
-  Coq and install the corresponding object files under ``.coq-native``, when in
-  the ``release`` profile. The regular ``dev`` profile skips native compilation
-  to make the build faster. This has been available since :ref:`Coq lang
```

Comment:
Since the behavior is kept, the sentence "The regular ``dev`` profile skips native compilation to make the build faster." should also be kept. It can be a pretty disturbing behavior, so I think it should at the very least be documented.

Reply:
Yeah I am not a fan of it at all TBH.

Reply:
Me neither, for the following reason, after this PR:
* most users will use the default (no `mode` stanza)
* when something is not compatible with coqnative, `mode vo` can be used
* so `mode native` has probably no other use than to test compilation with native, the more complex behavior where it is disabled in dev mode can then be more disturbing than anything for those users

(of course I might be missing something)

Reply:
I will remove it. If we want something like this in the future, we should support it correctly. Either we add the split native compilation which will allow users to compile native as a second step or we allow users to explicitly disable native themselves in their profiles.

But keeping this confusing default behavior isn't great. As you said, `(mode vo)` can always be used.

Reply:
`mode native` is indeed made obsolete by this PR. `dev` disabling native is indeed a mixed bag, and was also motivated by the lack of dynamic configuration, previosly you had to edit the stanza to disable native for example when developing Coq, as you don't want to take the expensive compilation route.

Now you can set it at configure time, so the tradeoff is a bit better.